src/services/router/router.tsx

```diff
@@ -0,0 +1,29 @@
+import { LoaderFunction, createBrowserRouter } from 'react-router-dom';
+import RootLayout from '../../layouts/RootLayout';
+import ErrorPage from '../../pages/ErrorPage/ErrorPage';
+import Home from '../../pages/Home/Home';
+import PlanetDetails from '../../components/PlanetDetails/PlanetDetails';
+import planetDetailsLoader from '../../components/PlanetDetails/planetDetailsLoader';
+
+const router = createBrowserRouter([
+  {
+    path: '/',
+    element: <RootLayout />,
+    errorElement: <ErrorPage />,
+    children: [
+      {
+        path: 'search/:pageNumber',
+        element: <Home />,
+        children: [
+          {
+            path: ':planetName',
+            element: <PlanetDetails />,
+            loader: planetDetailsLoader as LoaderFunction,
```

Comment:
avoid type casting

Reply:
TODO: i need some research on how to avoid type casting with Loaders